### ISSUE TITLE
[sysrst_ctrl] Unable to reset OT

### ISSUE THREAD

**Issue by jesultra** [lowRISC/opentitan#20392]
### Description

I am trying to program the sysrst controller in Google Ti50 codebase, and I am setting one key combination to trigger a pulse on EC reset output, and another combination to trigger OT reset.

I see a fine EC reset pulse, but I cannot get OT to reset.  To eliminate issues in my trigger conditions, I tried setting the `COM_OUT_CTL::INTERRUPT_0` instead, and if I do so, I do see an interrupt in response to my trigger condition.  But setting `COM_OUT_CTL::RST_REQ_0` seems to have no effect.

I am setting PWRB_IN_SEL_0, KEY0_IN_SEL_0, KEY2_IN_SEL_0 in COM_SEL_CTL and COM_DET_CTL to 2,500,000 for a 10-second trigger condition.

**Comment by pamaury** [comment]
Did you enable the sysrst control source in the power manager:
- https://opentitan.org/book/hw/top_earlgrey/ip_autogen/pwrmgr/doc/registers.html#reset_en
Probably something like:
```c
dif_pwrmgr_set_request_sources(&pwrmgr, kDifPwrmgrReqTypeReset,
  kDifPwrmgrResetRequestSourceOne, kDifToggleEnabled));
```
I was not able to find which source corresponds to which block in the documentation but there is a [DV test ](https://github.com/lowRISC/opentitan/blob/master/sw/device/tests/sim_dv/sysrst_ctrl_ec_rst_l_test.c#L133) that says it is one.

**Comment by a-will** [comment]
I don't think this is documented yet. For earlgrey's mapping...

https://github.com/lowRISC/opentitan/blob/b08261623f1f04a18f73de5ea713c622ebb18209/hw/top_earlgrey/sw/autogen/top_earlgrey.h#L1737-L1741

Just bear in mind that the DIFs have 1-based indexing (for names), whereas the enum values are zero-based.

**Comment by jesultra** [comment]
I had no idea that the reset signal from sysrst_ctrl went through the power manager, the documentation for sysrst does not mention this, but I guess the routing is a feature of the top-level design, (though maybe the sysrst documentation could say that the reset signal _might be_ futher gated in the power manager).  I will try enabling the low bit of RESET_EN, and see if that makes a difference, I expect it to.

**Comment by jesultra** [comment]
Does `//sw/device/tock:tock_lowrisc_registers` not generate a register declaration file for the power manager?  I do not see any of the file generated names start with "power" or "pwr".

**Comment by a-will** [comment]
> Does `//sw/device/tock:tock_lowrisc_registers` not generate a register declaration file for the power manager? I do not see any of the file generated names start with "power" or "pwr".

Nope! You're looking for `//sw/device/tock:tock_earlgrey_registers`, by the looks of it.

**Comment by jesultra** [comment]
Thanks for the pointers.

I tried writing 1 into the two lowest bits of RESET_EN, **and** then writing 1 to CFG_CDC_SYNC and waiting for it to become zero.  This seems to solve my issue, I now see the core being reset upon the configured keyboard combination being pressed.
